### ISSUE TITLE
Add support for random port allocation

### ISSUE THREAD

**Issue by oleg-nenashev** [quarkiverse/quarkus-wiremock#68]
The extension often fails in the test when trying to allocate a port from higher or lower range, It would be nice if the extension could pick a random free port on startup, e.g. when `random` passed in the configuration

## Sample output

```
....
Caused by: io.quarkus.runtime.QuarkusBindException: Port(s) already bound: 8081: Address already in use: bind
 at io.quarkus.vertx.http.runtime.VertxHttpRecorder$11.handle([VertxHttpRecorder.java:737](vscode-file://vscode-app/c:/Users/Oleg/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html))
 at io.quarkus.vertx.http.runtime.VertxHttpRecorder$11.handle([VertxHttpRecorder.java:718](vscode-file://vscode-app/c:/Users/Oleg/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html))
 at io.vertx.core.impl.future.FutureImpl$3.onFailure([FutureImpl.java:153](vscode-file://vscode-app/c:/Users/Oleg/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html))
 at io.vertx.core.impl.future.FutureBase.emitFailure([FutureBase.java:75](vscode-file://vscode-app/c:/Users/Oleg/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html))
 at io.vertx.core.impl.future.FutureImpl.tryFail([FutureImpl.java:230](vscode-file://vscode-app/c:/Users/Oleg/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html))
 at io.vertx.core.impl.future.Mapping.onFailure([Mapping.java:45](vscode-file://vscode-app/c:/Users/Oleg/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html))
```

**Comment by oleg-nenashev** [comment]
It's an RFE, so totally fine if it will not be implemented soon :) Quarkiverse - wide issue templates would be cool though 

It depends on the operating system, but I got it a few times on high-range ports. 

Apparently it also happens during live coding when service is reloaded, it seems there is a race condition on restart. It is not a huge deal but might confuse some users

**Comment by chberger** [comment]
> It's an RFE, so totally fine if it will not be implemented soon :) Quarkiverse - wide issue templates would be cool though
Yeah, I should define some templates ...
 
> It depends on the operating system, but I got it a few times on high-range ports.
> 
> Apparently it also happens during live coding when service is reloaded, it seems there is a race condition on restart. It is not a huge deal but might confuse some users

Ah, this sounds bad. I've implemented your RFE with https://github.com/quarkiverse/quarkus-wiremock/pull/74. Would you like to take a look at it?

**Comment by chberger** [comment]
@oleg-nenashev : Thanks for your review. Would you like to test the code from `main` or are you waiting for a released version?

**Comment by oleg-nenashev** [comment]
I will try with Main and create a downstream PR so that there is an
example, thanks!

**Comment by oleg-nenashev** [comment]
https://github.com/wiremock/wiremock-quarkus-devservice-demo/pull/1 verifies that. Thanks for the patch!

**Comment by chberger** [comment]
Feature has been released with [1.1.0](https://github.com/quarkiverse/quarkus-wiremock/releases/tag/1.1.0)
